process.c

```diff
@@ -4679,56 +4679,132 @@ rb_spawn(int argc, const VALUE *argv)
- *    command name and one or more arguments (no shell)
- *  [<code>[cmdname, argv0], arg1, ...</code>]
- *    command name, <code>argv[0]</code> and zero or more arguments (no shell)
+ *  Returns:
  *
- *  system returns +true+ if the command gives zero exit status,
- *  +false+ for non zero exit status.
- *  Returns +nil+ if command execution fails.
- *  An error status is available in <code>$?</code>.
+ *  - +true+ if the command exits with status zero.
+ *  - +false+ if the exit status is a non-zero integer.
+ *  - +nil+ if the command fails.
  *
- *  If the <code>exception: true</code> argument is passed, the method
- *  raises an exception instead of returning +false+ or +nil+.
+ *  Raises an exception (instead of returning +false+ or +nil+)
+ *  if keyword argument +exception+ is set to +true+.
  *
- *  The arguments are processed in the same way as
- *  for Kernel#spawn.
+ *  Assigns the command's error status to global variable <tt>$?</tt>.
  *
- *  The hash arguments, env and options, are same as #exec and #spawn.
- *  See Kernel#spawn for details.
+ *  The new process is created using the
+ *  {system system call}[https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/functions/system.html];
+ *  it may inherit some of its environment from the calling program
+ *  (possibly including open file descriptors).
  *
- *     system("echo *")
- *     system("echo", "*")
+ *  Argument +env+, if given, is a hash that affects +ENV+ for the new process;
+ *  see {Execution Environment}[rdoc-ref:Process@Execution+Environment].
  *
- *  <em>produces:</em>
+ *  Argument +options+ is a hash of options for the new process;
+ *  see {Execution Options}[rdoc-ref:Process@Execution+Options].
+ *
+ *  The first required argument is one of the following:
+ *
+ *  - +command_line+ if it is a string,
+ *    and if it begins with a shell reserved word or special built-in,
+ *    or if it contains one or more metacharacters.
+ *  - +exe_path+ otherwise.
+ *
+ *  <b>Argument +command_line+</b>
+ *
+ *  \String argument +command_line+ is a command line to be passed to a shell;
+ *  it must begin with a shell reserved word, begin with a special built-in,
+ *  or contain meta characters:
+ *
+ *    system('echo')                             # => true  # Built-in.
+ *    system('if true; then echo "Foo"; fi')     # => true  # Shell reserved word.
+ *    system('date > /tmp/date.tmp')             # => true  # Contains meta character.
+ *    system('date > /nop/date.tmp')             # => false
+ *    system('date > date.tmp', exception: true) # Raises RuntimeError.
+ *
+ *  Assigns the command’s error status to global variable <tt>$?</tt>:
+ *
+ *    system('echo')                             # => true  # Built-in.
+ *    $?                                         # => #<Process::Status: pid 640610 exit 0>
+ *    system('date > /nop/date.tmp')             # => false
+ *    $?                                         # => #<Process::Status: pid 640742 exit 2>
```

Comment:
These examples are under `Argument +command_line+` which makes it seem like `$?` is only set for the command line case and not for the executable path case. IMO we should just move this up, where we discuss `$?`.

Reply:
Mentioned also for executable path (ll. 4785 ff.).  The thing about the first mention, early in the general discussion (l. 4703), is that we haven't yet had any examples of any kind.  I think we need to mention at 4703, but give examples in the two later spots.